cmd/pipeline/execute.go

```diff
@@ -87,7 +87,7 @@ func executePipeline(cmd *cobra.Command, options *executeOptions) error {
 		}
 	}
 
-	resp, err := options.GateClient.PipelineControllerApi.InvokePipelineConfigUsingPOST1(options.GateClient.Context,
+	successPayload, resp, err := options.GateClient.PipelineControllerApi.InvokePipelineConfigUsingPOST1(options.GateClient.Context,
```

Comment:
Instead of doing the JSON decoding in the generated file, why not attempt a JSON decode of the response body `resp` here?

Reply:
It looks like it might be better to use the newer Gate API `options.GateClient.PipelineControllerApi.InvokePipelineConfigViaEchoUsingPOST` instead.

This returns an `interface{}` which we can marshal into a struct.
The struct will hold a list of executed pipelines because a single trigger (`spin`'s POST) can invoke multiple pipelines.

See Gate groovy returned object which would land in the Go empty interface.
```
    return [
      eventId: eventId,
      ref    : String.format("/pipelines/%s", executionId)
    ]
```
Per: https://github.com/spinnaker/gate/blob/6e751aa7d7c9a4f116b85b6f4b73d5203d902ce7/gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/PipelineService.groovy#L94-L116